code/android-core-library/src/phone/java/com/adobe/marketing/mobile/MobileCore.java

```diff
@@ -991,4 +992,9 @@ public static boolean dispatchResponseEvent(@NonNull final Event responseEvent,
         return true;
     }
 
+    @VisibleForTesting
+    public static void resetFlag() {
+        sdkInitializedWithContext.set(false);
```

Comment:
Can you just make `sdkInitializedWithContext` variable package private? 
Integration tests and other extensions should create a MobileCoreExt class in the com.adobe.marketing.mobile package to reset both this flag and EventHub.